pkg/controller/jobcontrol.go

```diff
@@ -244,6 +226,37 @@ func (c *jobControl) ControlJobs(
 		return JobControlDeletingJobs, nil, err
 	}
 
+	if currentJobName != "" {
```

Comment:
Do we even care about tracking current job any more? The only reason we tracked it previously was so that we could tell when a job was lost. If we are going to re-create the job even when it was lost, then I don't see much point in distinguish a lost job from a failed job.

Reply:
Yes, I agree we don't need it. However, it felt like a lot of changes all at once, but if you're fine with me removing it, I can proceed.

Reply:
I vote for removing it now. It is only going to make the code simpler.